test/E2ETest/Microsoft.Test.E2E.AspNet.OData/DollarQuery/DollarQueryTests.cs

```diff
@@ -0,0 +1,205 @@
+            }
+        }
+
+        protected override void UpdateConfiguration(WebRouteConfiguration configuration)
+        {
+            configuration.Routes.Clear();
+            configuration.Count().Filter().OrderBy().Expand().MaxTop(null).Select();
+            configuration.MapODataServiceRoute("odata", "odata", GetEdmModel(configuration), new DefaultODataPathHandler(), ODataRoutingConventions.CreateDefault());
+        }
+
+        private static IEdmModel GetEdmModel(WebRouteConfiguration configuration)
+        {
+            ODataConventionModelBuilder builder = configuration.CreateConventionModelBuilder();
+
+            EntitySetConfiguration<DollarQueryCustomer> dollarQueryCustomers =
+                builder.EntitySet<DollarQueryCustomer>("DollarQueryCustomers");
+
+            EntitySetConfiguration<DollarQueryOrder> dollarQueryOrders =
+                builder.EntitySet<DollarQueryOrder>("DollarQueryOrders");
+
+            return builder.GetEdmModel();
+        }
+
+        [Theory]
+        [MemberData(nameof(ODataQueryOptionsData))]
+        public async Task ODataQueryOptionsInRequestBody_ForSupportedMediaType(string resourcePath, string queryOptionsPayload)
+        {   
+            string requestUri = this.BaseAddress + resourcePath + "/$query";
+            var contentType = "text/plain";
+
+            var request = new HttpRequestMessage(HttpMethod.Post, requestUri);
+            request.Content = new StringContent(queryOptionsPayload);
+            request.Content.Headers.ContentType = new MediaTypeWithQualityHeaderValue(contentType);
+
+            request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse(ApplicationJsonODataMinimalMetadataStreamingTrue));
+
+            var response = await this.Client.SendAsync(request);
+
+            Assert.True(response.IsSuccessStatusCode);
+        }
+
+        [Fact]
+        public async Task ODataQueryOptionsInRequestBody_ReturnsExpectedResult()
+        {
+            string requestUri = this.BaseAddress + CustomersResourcePath + "/$query";
+            var contentType = "text/plain";
+            var queryOptionsPayload = "$filter=Id eq 1";
+
+            var request = new HttpRequestMessage(HttpMethod.Post, requestUri);
+            request.Content = new StringContent(queryOptionsPayload);
+            request.Content.Headers.ContentType = new MediaTypeWithQualityHeaderValue(contentType);
+
+            request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse(ApplicationJsonODataMinimalMetadataStreamingTrue));
+
+            var response = await this.Client.SendAsync(request);
+            Assert.True(response.IsSuccessStatusCode);
+
+            var contentAsString = response.Content.ReadAsStringAsync().Result;
+            Assert.Contains("\"value\":[{\"Id\":1,\"Name\":\"Customer Name 1\"}]", contentAsString);
+        }
+
+        [Fact]
+        public async Task ODataQueryOptionsInRequestBody_ForUnsupportedMediaType()
```

Comment:
>    public async Task ODataQueryOptionsInRequestBody_ForUnsupportedMediaType() [](start = 4, length = 78)

any similar tests for a media type supported through a custom queryoptionparser?

Reply:
Added a test to cover that scenario
https://github.com/OData/WebApi/pull/2191/files#diff-9e480fb8b0555db1128de67981ff5e0aR321
Here is the custom query options cursor for media type _text/xml_
https://github.com/OData/WebApi/pull/2191/files#diff-9e480fb8b0555db1128de67981ff5e0aR244

Reply:
nice!

---
In reply to: [488802104](https://github.com/OData/WebApi/pull/2191#discussion_r488802104) [](ancestors = 488802104)